Tasks/JavaToolInstallerV0/chocoJavaInstall.ps1

```diff
@@ -0,0 +1,89 @@
+param (
+	[string]$VersionSpec
+    )
+
+# Set new machine path 
+function Set-MachinePath{
+    [CmdletBinding()]
+    param(
+        [string]$NewPath
+    )
+    Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name Path -Value $NewPath
+    return $NewPath
+}
+
+# Get machine path from registry
+function Get-MachinePath{
+    [CmdletBinding()]
+    param(
+
+    )
+    $currentPath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).Path
+    return $currentPath
+}
+
+$jdkVersion = ("jdk" + $VersionSpec)
+$jdkMatchString = ('jdk*' + $VersionSpec + '*')
+$extendedJavaHome = ('JAVA_HOME_' + $VersionSpec + '_X64')
+
+Write-Host "Installing chocolatey"
+$chocoExePath = 'C:\ProgramData\Chocolatey\bin'
+
+if ($($env:Path).ToLower().Contains($($chocoExePath).ToLower())) {
```

Comment:
Is this what we do in other installers? Seems like we should just run the equivalent of `where choco`. Does the vsts Powershell library include a which/where function?

Reply:
I would split this into two files - chocolatelyInstall.ps1 and chocoJavaInstall.ps1. Your typescript code can run `which('choco')` and call chocolatelyInstall.ps1 if necessary. Then it can call chocoJavaInstall.ps1. Later we can reuse chocolatelyInstall.ps1 elsewhere without modification.